ledger/participant-integration-api/src/main/scala/platform/store/backend/common/UserManagementStorageBackendTemplate.scala

```diff
@@ -180,25 +173,25 @@ object UserManagementStorageBackendTemplate extends UserManagementStorageBackend
     }
   }
 
-  private def fromUserRight(right: UserRight): (Int, String) = {
+  private def fromUserRight(right: UserRight): (Int, Option[Ref.Party]) = {
     right match {
-      case ParticipantAdmin => (Right.PARTICIPANT_ADMIN_FIELD_NUMBER, AbsenceOfPartyMarker)
-      case CanActAs(party) => (Right.CAN_ACT_AS_FIELD_NUMBER, party: String)
-      case CanReadAs(party) => (Right.CAN_READ_AS_FIELD_NUMBER, party: String)
+      case ParticipantAdmin => (Right.PARTICIPANT_ADMIN_FIELD_NUMBER, None)
+      case CanActAs(party) => (Right.CAN_ACT_AS_FIELD_NUMBER, Some(party))
+      case CanReadAs(party) => (Right.CAN_READ_AS_FIELD_NUMBER, Some(party))
       case _ =>
         throw new RuntimeException(s"Could not recognize user right: $right!")
     }
   }
 
-  private def dbStringToPartyString(raw: String): Option[Ref.Party] = {
-    if (raw == AbsenceOfPartyMarker)
-      None
-    else
-      Some(Ref.Party.assertFromString(raw))
+  private def dbStringToPartyString(raw: Option[String]): Option[Ref.Party] = {
+    raw.map(Ref.Party.assertFromString)
   }
 
-  private def partyToDbString(party: Option[Ref.Party]): String = {
-    (party: Option[String]).getOrElse(AbsenceOfPartyMarker)
+  private def isForPartyPredicate(forParty: Option[Ref.Party]): ComposableQuery.CompositeSql = {
+    import com.daml.platform.store.backend.common.ComposableQuery.SqlStringInterpolation
+    forParty.fold(cSQL"IS NULL") { party: Ref.Party =>
```

Comment:
For additional safety, you could add the following line here, because comparison with an empty string would not work as expected on Oracle:
```
require(party.length > 0, "party name must be non-empty")
```

Reply:
That's already implied by accepting `Ref.Party` type which guarantees the string is not empty (com.daml.lf.data.IdStringImpl#Party).
Do you think it's still worth adding an extra check to be extra safe?

Reply:
No need to be extra safe when it's not necessary.